codex/stores/networkstore.nim

```diff
@@ -63,6 +64,22 @@ method putBlock*(
   await self.engine.resolveBlocks(@[blk])
   return success()
 
+method ensureExpiry*(
+    self: NetworkStore,
+    cid: Cid,
+    expiry: SecondsSince1970
+): Future[?!void] {.async.} =
+  ## Ensure that block's assosicated expiry is at least given timestamp
+  ## If the current expiry is lower then it is updated to the given one, otherwise it is left intact
+  ##
+
+  if (await self.localStore.hasBlock(cid)).tryGet:
+    return await self.localStore.ensureExpiry(cid, expiry)
+  else:
+    trace "Updating expiry - block not in local store", cid
```

Comment:
I disagree. If this is called with CID of not locally stored block, then it will create an expiry entry for a non-existing block, which might lead to potential unexpected behavior. Will that happen? Most probably not, but still, this is IMHO safer approach.

Reply:
My previous comment is bit wrong in the point of creating expiry for non-existing blocks - that should not happen. But I still think that this part of the Network Store implementation is correct.

@benbierens, what do you mean by "handle non-existing blocks"? Fail or ignore for blocks not found?

IMHO in NetworkStore we should ignore non-local (e.g. non-existing) blocks, but in the context of LocalStore IMHO we should fail for non-existing blocks... WDYT?

Reply:
As stated previously, I don't think there's a nice implementation of this method in the network store, as we cannot modify what's stored in the network, only locally.

Reply:
Sooo I keep it? 😅

Reply:
You can keep it I think :) Maybe we will remove it in the future when we decide to decouple networkstore from the blockstore interface.

Reply:
We have noticed in several places that the blockstore interface is causing us trouble and not giving us any or enough benefits. So it should go. I agree there's no nice way to implement this.
The repostore implementation though should definitely not allow the creation of metadata for non-existing blocks.